File: scripts/algorithms/F/Fizz Buzz/Fizz Buzz.java
class Solution {
    public List<String> fizzBuzz(int n) {
        List<String> l=new ArrayList<>();
        
        for(int i=1,fizz=0,buzz=0;i<=n;i++)
        {
            fizz++;
            buzz++;
            
            if(fizz==3 && buzz==5)
            {
                l.add("FizzBuzz");
                fizz=0;
                buzz=0;
            }
            else if(fizz==3)
            {
                l.add("Fizz");
                fizz=0;
            }
            else if(buzz==5)
            {
                l.add("Buzz");
                buzz=0;
            }
            else{
                l.add(String.valueOf(i));
            }
        }
        return l;
    }
}

File: scripts/algorithms/F/Fizz Buzz/Fizz Buzz.cpp
// Runtime: 8 ms (Top 29.27%) | Memory: 8 MB (Top 33.77%)
class Solution {
public:
    vector<string> fizzBuzz(int n) {
        vector<string> ans;
        string hehe;
        for (int i = 1; i <= n; i++) {
            if (i % 3 == 0 and i % 5 == 0) hehe += "FizzBuzz";
            else if (i % 3 == 0) hehe += "Fizz";
            else if (i % 5 == 0) hehe += "Buzz";
            else hehe = to_string(i);
            ans.push_back(hehe);
            hehe = "";
        }
        return ans;
    }
};